### DISCUSSION TITLE
About the RP2040 and 16-bit parallel display

### DISCUSSION THREAD

**Discussion by rtek1000** [Bodmer/TFT_eSPI#3189] [Q&A - General]
I saw about the RP2040 being able to use 16-bit DMA. Is the performance of the RP2040 (133 MHz) using 16-bit DMA better than the ESP32 (240MHz)  8-bit no DMA?

Is it possible to choose the 16-bit bus operating mode?

The SSD1963 controller datasheet shows 2 possible uses, RGB888 and RGB565. But it should be perhaps 3x faster to use RGB565, when the data is sent in just one installment, while for RGB888 there are 3 data sends.

- A note: when using [RGB565](https://barth-dev.de/online/rgb565-color-picker/), it may be necessary to use an RGB888 color table for screen capture, otherwise it is not possible to obtain exactly the same colors when viewing the captured RGB888 image (I had to do this in a [project](https://github.com/rtek1000/Datalogger_2039) where the screen is drawn pixel by pixel), I found some ways to convert RGB565 to RGB888, but the colors are never faithfully recovered.

![SSD1963_16bits](https://github.com/Bodmer/TFT_eSPI/assets/8660811/be0bfe2d-6a4d-4f34-b174-849613b0605b)

**Comment by PJ5168** [comment]
Where is the memory?
SSD1963 800x480 RGR565 = 800x480x2 = 768KB ! But the RP2040 only has 262KB.
If you want to have framebuffer or sprite, 1.5MB is needed. No way for RP2040.
The QSPI flash is slow. And 4-bit SDIO is slow as well.

The ESP32 S3 can use 8MB 120Mhz Oct. SPI PSRAM as framebuffer. Then the 800x480x2 16bit parallel is about 41FPS.
